### DISCUSSION TITLE
Best Authntication workflow

### DISCUSSION THREAD

**Discussion by mo-fouad** [nandorojo/solito#87] [General]
Hello @nandorojo 
Awesome work, I got inspired by your talk and decided to change my workflow to follow your steps, I read your comments on the previous discussion regarding Authentication generally.
I tried to use the [firebase](https://github.com/firebase/firebase-js-sdk) SDK to create the authentication worked for the web app but did not work on mobile apps. 
tried to use [React Native Firebase](https://rnfirebase.io/) worked for Mobile but unfortunately did not work for the Web app. 

from your perspective, what would be the best practice here to achieve the authentication on both apps?
also if I want to do some logic inside every component based on the use itself, how it can be done in an effective way?

Thanks a lot ^^

**Comment by nandorojo** [comment]
thanks, happy to hear that!

yeah, we use RNFB on native and normal firebase on web by using a `index.web.tsx` file, and exporting commonly used functions that match up, such as `getUser()`, `onIdTokenChanged`, etc
